dogen/tools.py

```diff
@@ -87,6 +87,11 @@ def fetch(self):
         """ Fetches the artifact to the artifact dir """
         self._generate_url()
         destination = os.path.join(self.target_dir, self.name)
+        # If the artifacts exist just return - we dont care if sum is correct here
+        if os.path.exists(destination):
```

Comment:
Where do we check the hash of the file available locally?

Reply:
in verify() 
its called after fetch as you can see here:
https://github.com/jboss-dockerfiles/dogen/blob/dogen-ng/dogen/generator.py#L69